### PR TITLE
DAOS-9089 utils: re-enable autotest punch tests

### PR DESCRIPTION
**Pull request by wangshilong**
A couple of improvements:

1. Enable punch test and fixed assertion because we don't pass correct object handle
   to kv_punch()
2. Fixed to return correct error in some cases.
3. limit write by 1/10 available free space so that pool size doesn't depend on hardware
performance.

Signed-off-by: Wang Shilong <shilong.wang@intel.com>

### CODE REVIEW

**Comment by johannlombardi** [comment]
Maybe use a commit pragma just to run this test? That should be way faster.

**Comment by johannlombardi** [comment]
Merged master again.

File: src/utils/daos_autotest.c [hunk 1]
```diff
@@ -709,7 +750,6 @@ kv_read128(void)
 	return 0;
 }
 
-#if 0
```
Comment (johannlombardi) [inline src/utils/daos_autotest.c:712]: Comment below should be removed too (can be done in another PR).

**Review by johannlombardi — APPROVED** [review]

File: src/utils/daos_autotest.c [hunk 2]
```diff
@@ -709,7 +750,6 @@ kv_read128(void)
...
 /**
  * Disable since it triggers an assertion error on the client.
```
Comment (bfaccini) [inline src/utils/daos_autotest.c:754]: This comment must be removed now

File: src/utils/daos_autotest.c [hunk 3]
```diff
@@ -727,12 +767,13 @@ kv_punch(void)
 		return rc;
 	}
 
-	punch_rc = daos_obj_punch(oh, DAOS_TX_NONE, 0, NULL);
+	punch_rc = daos_obj_punch(daos_kv2objhandle(oh), DAOS_TX_NONE,
```
Comment (bfaccini) [inline src/utils/daos_autotest.c:770]: Is it guaranteed that the API will gracefully handle DAOS_HDL_INVAL return from daos_kv2objhandle() ?

Comment (wangshilong) [reply src/utils/daos_autotest.c:770]: Agreed, bttter add check, and addressed above johan's comments as well.

**Review by bfaccini — APPROVED** [review]

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**